src/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,70 @@
+import {getDocument, getSetImmediate, newMutationObserver} from './helpers'
+
+function waitForElementToBeRemoved(
+  callback,
+  {
+    container = getDocument(),
+    timeout = 4500,
+    mutationObserverOptions = {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      characterData: true,
+    },
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    if (typeof callback !== 'function') {
+      reject(
+        new Error(
+          'waitForElementToBeRemoved requires a function as the first parameter',
+        ),
+      )
+    }
+
+    // Check if the element is not present synchronously,
+    // As the name waitForElementToBeRemoved should check `present` --> `removed`
+    try {
+      const result = callback()
+      if (!result || (Array.isArray(result) && !result.length)) {
+        onDone(
+          new Error(
+            'The callback function which was passed did not return an element or non-empty array of elements. waitForElementToBeRemoved requires that the element(s) exist before waiting for removal.',
+          ),
+        )
+      }
+    } catch (error) {
+      onDone(error)
+    }
+
+    const timer = setTimeout(onTimeout, timeout)
+    const observer = newMutationObserver(onMutation)
+    observer.observe(container, mutationObserverOptions)
+    function onDone(error, result) {
+      const setImmediate = getSetImmediate()
+      clearTimeout(timer)
```

Comment:
`onDone` can be called before `timer` is even defined, which results in an error. Make sure to define `timer` earlier in the function.